src/bionics.cpp

```diff
@@ -1536,13 +1545,31 @@ void load_bionic(JsonObject &jsobj)
     if (faulty) {
         faulty_bionics.push_back(id);
     }
-    if (power_source) {
-        power_source_bionics.push_back(id);
-    }
-    // Things are activatable if they are toggleable, have a activation cost, or have a activation time
-    if (!toggled && on_cost <= 0 && time <= 0) {
-        unpowered_bionics.push_back(id);
+
+    auto const result = bionics.insert(std::make_pair(std::move(id),
+        bionic_data(std::move(name), power_source, toggled, on_cost, off_cost, react_cost, time,
+                    std::move(description), faulty)));
+
+    if (!result.second) {
+        debugmsg("duplicate bionic id");
     }
+}
+
+void bionic::serialize(JsonOut &json) const
```

Comment:
Those functions usually go into savegame_json.cpp, but this is not done consistently anyway and I myself don't like that separation either, so it's fine.